.github/workflows/rust.yml

```diff
@@ -1,39 +1,75 @@
 name: Rust tests
+
 on:
   push:
     branches:
-      - main
-  pull_request:
-
-env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  rust-build-test:
+  rust-test:
     runs-on: ubuntu-20.04
     steps:
       - name: Checkout branch
         uses: actions/checkout@v2
+      - name: Set up Rust caches
+        uses: actions/cache@v2
+        id: rust-cache
         with:
-          profile: minimal
-          toolchain: stable
-          override: true
+          path: |
+            ~/.cargo/bin
+            ~/.cargo/registry/index
+            ~/.cargo/registry/cache
+            ~/.cargo/git/db
+            target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+      - name: Run steward acceptance tests
+        run: cargo test --all --verbose
+
+  rust-build:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v2
       - name: Set up Rust caches
-        uses: actions/cache@v3
+        uses: actions/cache@v2
         id: rust-cache
         with:
           path: |
-            ~/.cargo/registry
-            ~/.cargo/git
+            ~/.cargo/bin
+            ~/.cargo/registry/index
+            ~/.cargo/registry/cache
+            ~/.cargo/git/db
             target
           key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+      - name: Run Cargo build
+        run: cargo build
+
+  fmt:
+    name: Rustfmt
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v2
+        with:
+          profile: minimal
+          toolchain: stable
```

Comment:
This action uses `with: profile: toolchain:`, as does `clippy`, but the others do not. We should either explicitly include them all using the toolchain of the rust version we expect rather than `stable`, or we should remove them all if it turns out they are unnecessary (they might have been deperecated?).